Source/msg.cpp

```diff
@@ -2389,10 +2334,16 @@ size_t OnSpawnMonster(const TCmd *pCmd, const Player &player)
 
 	size_t typeIndex = static_cast<size_t>(SDL_SwapLE16(message.typeIndex));
 	size_t monsterId = static_cast<size_t>(SDL_SwapLE16(message.monsterId));
+	uint8_t golemOwnerPlayerId = message.golemOwnerPlayerId;
+	int16_t golemSpellLevel = SDL_SwapLE16(message.golemSpellLevel);

```

Comment:
Maybe we should not trust whatever spell level the remote player sends us. #7485

Reply:
Good idea. I added a validation to the spell level 🙂 . 

Unfortunately I don't think I can do it like in #7485 because `AddGolem` happens when the spell is cast and then we send the network message. By the time the message arrives on other clients, they may already have another spell in the queue. But maybe I am missing something.

Reply:
I didn't want to hardcode the golem spell too much. My thought was that a modder who wants to change the behaviour in the future (add a second type of golem) doesn't need to change the logic here.
But I'm also happy with a change to `player.GetSpellLevel(SpellID::Golem)`. 🙂